samples/samples.jl

```diff
@@ -0,0 +1,213 @@
+# low-level wrappers of the MAGMA library
+include("common.jl")
+
+
+# magma_init
+function magmaInit()
+	ccall((:magma_init, libmagma),Cint,())
```

Comment:
So after you move these implementation to `src`, we need to have some unit tests that make sure each function works or not. For working tests you can use `@test` to mark them
for broken tests use `@test_broken` for temporary.

Reply:
OK, I will immediately go check some testing examples in the official Julia documentation.

Reply:
Do you have any clue where and how the GPU buffer error could ever come?
It's so weird. I don't think I have met with it one month ago...
BTW yesterday I opened a new branch in my repository for debugging this problem.
I checked all the results on Google but no one situation really matches with ours...
Maybe later I should ask on slack

ERROR: LoadError: ArgumentError: cannot take the CPU address of a GPU buffer
Stacktrace:
 [1] convert(::Type{Ptr{Float32}}, ::CUDAdrv.Mem.DeviceBuffer) at /home/jerry/.julia/packages/CUDAdrv/WVU1H/src/memory.jl:83
 [2] unsafe_convert(::Type{Ptr{Float32}}, ::CUDAdrv.Mem.DeviceBuffer) at /home/jerry/.julia/packages/CUDAdrv/WVU1H/src/memory.jl:32
 [3] unsafe_convert(::Type{PtrOrCuPtr{Float32}}, ::CUDAdrv.Mem.DeviceBuffer) at /home/jerry/.julia/packages/CUDAdrv/WVU1H/src/pointer.jl:115
 [4] magmaSgesvd(::Int64, ::Int64, ::Int64, ::Int64, ::CuArray{Float32,2}, ::Int64, ::CuArray{Float32,2}, ::CuArray{Float32,2}, ::Int64, ::CuArray{Float32,2}, ::Int64, ::CuArray{Float32,1}, ::Int64, ::CuArray{Float32,1}) at /home/jerry/MAGMA.jl-1/samples/MAGMA.jl/src/MAGMA.jl:49
 [5] gesvd!(::Int64, ::Int64, ::CuArray{Float32,2}, ::Int64, ::Int64, ::Int64) at /home/jerry/MAGMA.jl-1/samples/MAGMA.jl/src/MAGMA.jl:174
 [6] top-level scope at none:0
in expression starting at /home/jerry/MAGMA.jl-1/samples/test/test.jl:30